### PR TITLE
Fixes RAGIN' MAGES

### PR DESCRIPTION
**Pull request by Crazylemon64**
Re-implements the Ragin' Mages gamemode. I'd like to test this, though, to make sure everything's working right.
Associated with: #3155

### CODE REVIEW

File: code/game/gamemodes/wizard/raginmages.dm [hunk 1]
```diff
@@ -3,23 +3,25 @@
...
+
+/datum/game_mode/wizard/raginmages/pre_setup()
+	. = ..()
 	if(!max_mages)
-		for(var/mob/living/player in mob_list)
-			if (player.client && player.stat != DEAD)
-				playercount += 1
-			max_mages = round(playercount / 5)
+		//max_mages = round(num_players() / players_per_mage)
+		max_mages = 5
```
Comment (Crazylemon64) [inline code/game/gamemodes/wizard/raginmages.dm]: oopsy left in a debug

**Comment by Crazylemon64** [comment]
@Tastyfish @tigercat2000 @Fox-McCloud @Aurorablade @KasparoVy @FalseIncarnate  Alright, I've got this ready to test out. When's a good time to see if this works?

**Comment by tigercat2000** [comment]
@Crazylemon64 Getting all of us while we are awake and not busy is probably difficult, might be better off with getting one or two of us and asking for help from the IRC/teamspeak/OOC on the server

**Comment by Crazylemon64** [comment]
That is fair enough - though I also lack control over ports on my network, so... This worked with just the 2 players I have to test with, should we just try merging this and fixing bugs then?

**Comment by Fox-McCloud** [comment]
@Crazylemon64 

You can PM me on the forums for proper contact information.

**Comment by Fox-McCloud** [comment]
WTF modifier, because that's Ragin' Mages in a nutshell.

**Comment by Crazylemon64** [comment]
@Fox-McCloud I've messaged you forumside.

**Comment by Aurorablade** [comment]
ode/game/gamemodes/wizard/raginmages.dm:43:error: wizard.current.get_area: undefined proc
code/game/gamemodes/wizard/raginmages.dm:50:error: wizard.current.get_area: undefined proc
code/game/gamemodes/wizard/raginmages.dm:86:error: H.qdel: undefined proc

qdel(H)

dunno about the pother two.

**Comment by Crazylemon64** [comment]
Oops, I did some quick tweaks in the git web interface

When I get to my good computer, I'll get something solid going, and squash all the extra commits

**Comment by Crazylemon64** [comment]
Alright this works for every bit that I've tested -
* Spawning new wizards in from the ghost pool
* Creating the ghost-wizard from the client's prefs, so the ragin' mages more accurately represent a gaggle of regular wizards
* Scouring out all cliented/minded mobs from the wizard lair if some pile of garbage attempts to farm other wizards for their spellbooks

I also added a (probably hideous) proc to human.dm to create a human mob from a client's preferences, in order to accommodate my changes. I found it strange that no such proc existed before, so I copied much of the functionality over from new_player, and made the calls within generic.

Overall, I'm done tweaking this PR, though perhaps it needs a final round of tests? Well, I suppose you can merge it and see what I missed then once review is finished, it should otherwise work fine.

**Comment by Fox-McCloud** [comment]
https://github.com/tgstation/-tg-station/blob/master/code/game/gamemodes/wizard/raginmages.dm may be relevant, not sure--s'where the game mode came from.

**Pull request merged**